### DISCUSSION TITLE
Question regarding routes and i18n

### DISCUSSION THREAD

**Discussion by emilforsmann** [vercel/next.js#12172] [Help] [answered]
Hi there, 

I'm currently setting up localization and have already shared some links prior to setting up localization. 
Do you guys know if it's possible to always push "/en" in front of any requested URL?

so if I hit **/?id=123123123132** it will push **"/en/?id="123123123132"**
and if i hit  **"/en/?id="123123123132"** it will stay on  **"/en/?id="123123123132"**

Bonus questions,
Do you guys know the right way to redirect old link with query params to a new URL either through nextjs or through Zeit (Vercel).

**Comment by waptik** [comment, marked as answer]
For both questions asked, I think you need nextjs' `experimental.rewrites`. Check [this](https://github.com/zeit/next.js/discussions/11687) and [this](https://github.com/zeit/next.js/discussions/10724#discussioncomment-703) and [this(for locales)](https://github.com/zeit/next.js/discussions/11859)  examples.

**Comment by leerob** [comment]
Check out the new [i18n RFC](https://github.com/vercel/next.js/discussions/17078) and give us your feedback! 🚀
